jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/SetupAuthenticationDialog.kt

```diff
@@ -263,6 +263,9 @@ class SetupAuthenticationDialog(
         }
 
         SsoSessionConfigurationManager().updateSsoSessionProfileToConfigFile(
+            "${state.idcTabState.profileName}-" +
+                "${state.idcTabState.rolePopupState.roleInfo?.accountId()}-" +
+                "${state.idcTabState.rolePopupState.roleInfo?.roleName()}",
```

Comment:
why not string substitution?

Reply:
String substitution was initially given for readability ```"${state.idcTabState.profileName}-${state.idcTabState.rolePopupState.roleInfo?.accountId()}-${state.idcTabState.rolePopupState.roleInfo?.roleName()}"```
However, when running Detekt, it failed due to exceeding the maxLength constraint. In order to comply with this constraint, we had to resort to string concatenation

Reply:
the max line length lint rule does not apply for triple-quoted strings.
in general you should override the linter if it is forcing a change that is less readable